torch/csrc/tensor/python_tensor.cpp

```
@@ -405,7 +405,11 @@ void py_set_default_dtype(PyObject* obj) {
       [backend, scalar_type](PyTensorType *x) {
         return x->get_backend() == backend && x->get_scalar_type() == scalar_type;
       });
-    set_default_tensor_type(*it);
+    if (it != tensor_types.end()) {
+      set_default_tensor_type(*it);
+    } else {
+      throw TypeError("tensor type not found");
```

we should have the same error message as below `"invalid dtype object"`

---

maybe consolidate the logic to throw error for invalid dtype:

```
bool not_valid_dtype = true;
if (THPDtype_Check(obj)) {
    ...
    if (it != tensor_types.end()) {
       not_valid_dtype = false;
      set_default_tensor_type(*it);
    }
}
if (not_valid_dtype) throw TypeError("tensor type not found");
```